elk/extraction/prompt_dataset.py

```diff
@@ -256,3 +261,37 @@ def num_classes(self) -> int:
                 f"{self.label_column} of type "
                 f"{self.active_split.features[self.label_column]}"
             )
+
+
+class InterleavedDatasets(TorchDataset):
+    def __init__(
+        self,
+        datasets: list[PromptDataset],
+    ):
+        """
+        Interleave several (PromptDataset) datasets into a single dataset,
+        alternating between the datasets.
+        Only samples as many datapoints from each dataset as the smallest dataset.
+        Args:
+            datasets (`List[PromptDataset]`):
+                List of datasets to interleave.
+        """
+        self.datasets = datasets
+
+        if not datasets:
+            raise ValueError("Unable to interleave an empty list of datasets.")
+
+        lengths = [len(dset) for dset in datasets]
+        self.min_dataset_length = min(lengths)
+        self.num_datasets = len(datasets)
+
+    def __getitem__(self, index: int) -> list[Prompt]:
+        which_dataset = index % self.num_datasets
```

Comment:
a more kickass way of doing this is with `divmod`:
`which_elem, which_dataset = divmod(index, self.num_datasets)`